### ISSUE TITLE
Have gli_strict_warning take a format string

### ISSUE THREAD

**Issue by cspiegel** [garglk/garglk#355]
Currently the `gli_strict_warning` function takes a single argument, which is a string.  A few callers have to build up a string with `sprintf` before sending it to `gli_strict_warning`.  `gli_strict_warning` should instead just take a format string.

**Comment by curiousdannii** [comment]
This doesn't directly connect with what you've raised here, but I've thought for a while that it would be good to have some glk functions for warnings etc which go outside the standard UI system. I thought that it would be helpful to have functions that take not just a string, but that open a stream for flexible printing. I haven't got around to adding these functions to anything, but I've been thinking about how to add them to glkote or remglk.

http://curiousdannii.github.io/if/outofband.html

If you're thinking about changing garglk's warning system, do you think something like this would help?

**Comment by cspiegel** [comment]
Sorry, I missed this comment earlier.

Gargoyle's current warning system, such as it is, just writes to standard error.  The number of people who start gargoyle from the terminal, and who will thus see the output, is probably near zero.  As such, a logging system which is actually accessible to the user would probably be very helpful.  For Gargoyle itself, it'd require putting some thought into how to classify the current error messages: there are over 200, and I'd hazard a guess that the vast majorty wouldn't need to be seen by the user in the general case.  But that looks like the point of `severity_Log`.

In short, I'd be interested in seeing how we might integrate such a system into Glk/Gargoyle, yes.  At this point Gargoyle isn't particularly user-friendly, and it'd be nice to start lurching toward becoming more so.

**Comment by cspiegel** [comment]
Done in #361.
